### PR TITLE
chore(deps): add useful pytest-* plugins

### PR DESCRIPTION
**Pull request by teocns**
# 🔍 Review Summary 
 **Purpose**:
  Enhance testing capabilities by adding pytest plugins to improve dependency management, asynchronous testing, HTTP request recording, and mocking functionalities.

**Key Changes**:
  - **Enhancement**: Added `pytest-depends`, `pytest-asyncio`, `pytest-vcr`, and `pytest-mock` to the `dev` optional dependencies for improved testing capabilities.

**Impact**:
  Improves the efficiency and coverage of the testing process, ensuring more robust and reliable code. 

<details><summary>Original Description</summary>

No existing description found

</details>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,6 +29,12 @@ dependencies = [
 [project.optional-dependencies]
 dev = [
     "pytest==7.4.0",
+    "pytest-depends",
+    "pytest-asyncio",
+    "pytest-vcr",
+    "pytest-mock",
+    "pyfakefs",
+    # Soon to come: pytest-asyncio
```
Comment (areibman) [inline pyproject.toml]: Remove the comment

**Comment by areibman** [comment]
Are these dependencies tied to anything specifically? We should include them on a per-issue basis

**Comment by teocns** [comment]
Pretty standard-use libs, commonly used and proven to be required (except asyncio AFAIK), at least not until now

**Comment by areibman** [comment]
Merging since we've updated the contributor files

**Review by areibman — APPROVED** [review]

**Pull request merged**